### PR TITLE
Parse multipart bodies on demand and answer multipart parse failures with 413 or 400

### PR DESCRIPTION
**Pull request by rymsha**
## Summary

The dispatch servlet carries the multipart configuration and `RequestSerializer` called `getParameterMap()` for every request. For a `multipart/form-data` body that call makes Jetty parse the whole body up front, spooling parts above 10 KiB to the temp directory with 15 GiB limits, before any handler has decided it needs the parts. An anonymous client can keep the disk busy on any URL.

## Changes

- `RequestSerializer` no longer asks the container for parameters when the request carries any `multipart/*` body, detected with Guava's `MediaType`. Parameters then come from the query string only. Multipart bodies are parsed on demand by `MultipartService`, which is what `portal.getMultipartForm()` and the admin REST readers already use.
- `MultipartServiceImpl` answers parse problems with client errors instead of 500. A malformed `Content-Type` is treated as not multipart, a missing `boundary` parameter is a 400, the Servlet API's `IllegalStateException` for exceeded `maxFileSize`/`maxRequestSize` becomes 413, and a malformed body becomes 400.

## Behaviour change to be aware of

Text fields of a `multipart/form-data` POST used to show up in `req.params` because Jetty merged them into the parameter map. After this change `req.params` for multipart requests holds query parameters only. Controllers read form fields through `portal.getMultipartForm()`.

## Not changed

- `multipart.maxFileSize` and `multipart.maxRequestSize` keep their 15 GiB defaults. Lowering them is a product decision.
- `multipart.store` still defaults to `java.io.tmpdir`. The launcher points that at `$XP_HOME/work`, so the spool directory is not the shared system temp directory.

## Tests

- `RequestSerializerMultipartTest`: query-only parameters for `multipart/form-data` and `multipart/mixed`, malformed query string, form-urlencoded still using the parameter map, media type detection for form-data, mixed, related, byteranges, malformed and non-multipart types.
- `MultipartServiceImplTest`: malformed content type, missing boundary, limit exceeded, malformed body, servlet exception.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01NF5h4qbEHLSwEBTxQKSoMV

### CODE REVIEW

File: modules/web/web-impl/src/main/java/com/enonic/xp/web/impl/serializer/RequestSerializer.java [hunk 1]
```diff
@@ -62,9 +71,44 @@ private void setCookies( final HttpServletRequest from, final WebRequest to )
...
         for ( final Map.Entry<String, String[]> entry : from.getParameterMap().entrySet() )
         {
             to.getParams().putAll( entry.getKey(), Arrays.asList( entry.getValue() ) );
         }
     }
+
+    static boolean isMultipartFormData( final String contentType )
+    {
+        return contentType != null && contentType.regionMatches( true, 0, MULTIPART_FORM_DATA, 0, MULTIPART_FORM_DATA.length() );
+    }
```
[the remark this answers is not in the document]
Comment (rymsha) [reply modules/web/web-impl/src/main/java/com/enonic/xp/web/impl/serializer/RequestSerializer.java]: Fixed in f5eb751: the media type is cut at the first `;`, trimmed and compared case-insensitively for equality, so `multipart/form-data2` and `multipart/mixed` are no longer treated as multipart forms. Test `isMultipartFormData` covers the boundary cases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: modules/web/web-impl/src/main/java/com/enonic/xp/web/impl/serializer/RequestSerializer.java [hunk 2]
```diff
@@ -76,9 +84,50 @@ private void setCookies( final HttpServletRequest from, final WebRequest to )
...
     }
+
+    static boolean isMultipart( final String contentType )
+    {
+        if ( contentType == null )
+        {
+            return false;
+        }
+        final int parameters = contentType.indexOf( ';' );
+        final String mediaType = ( parameters < 0 ? contentType : contentType.substring( 0, parameters ) ).trim();
```
Comment (rymsha) [inline modules/web/web-impl/src/main/java/com/enonic/xp/web/impl/serializer/RequestSerializer.java]: @copilot substring is inefficient. We use region matches

**Comment by rymsha** [comment]
Closed without merging. This was a workaround at the point where `getParameterMap()` triggers Jetty's multipart parse; the root causes (eager parameter parsing, a multipart configuration on the catch-all servlet, 15 GiB defaults) are tracked in #12368 instead.

---
_Generated by [Claude Code](https://claude.ai/code)_
